### ISSUE TITLE
[Proposal]: [Dot object deconstruction]

### ISSUE THREAD

**Issue by fedeAlterio** [dotnet/csharplang#6197]
# Dot object deconstruction

* [x] Proposed
* [ ] Prototype: Not Started
* [ ] Implementation: Not Started
* [ ] Specification: Not Started

## Summary
[summary]: #summary

Allow creating a tuple from an object simply using the dot (.) operator.

## Motivation
The main motivation is lighter syntax and better readability.
Consider:

`var (from, to) = (request.LikeMessage.FromUsername, request.LikeMessage.ToUsername);`

The idea is to pull the common part behind the dot (.).

`var (from, to) = request.LikeMessage.(FromUsername, ToUsername);`

## Detailed design
Basically is a syntactic sugar:
`var (from, to) = request.LikeMessage.(FromUsername, ToUsername);`
should be lowered to:
```
var tmp = request.LikeMessage; 
var (from, to) = (tmp.FromUsername, tmp.ToUsername);
```

## Drawbacks
The main drawback is the similarity of the expression to the function call operator.

## Alternatives
An alternative is to change the syntax to:
`var (from, to) = request.LikeMessage.{FromUsername, ToUsername};`

**Comment by 333fred** [comment]
As per our README, all new items should start as discussions, not issues, so I'm converting this over.
